### PR TITLE
update hardware-and-software-requirements.md for v8.5

### PR DESCRIPTION
**Pull request by qiancai**
### What is changed, added or deleted? (Required)

Update the hardware-and-software-requirements.md according to https://github.com/pingcap/docs-cn/blob/release-8.1/hardware-and-software-requirements.md

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [ ] master (the latest development version)
- [ ] v8.5 (TiDB 8.5 versions)
- [ ] v8.4 (TiDB 8.4 versions)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/19184
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by Oreoxmt — APPROVED** [review]
LGTM

File: hardware-and-software-requirements.md [hunk 1]
```diff
@@ -17,85 +17,108 @@ As an open-source distributed SQL database with high performance, TiDB can be de
...
+        <td>openEuler 22.03 LTS SP1/SP3</td>
+        <td><ul><li>x86_64</li><li>ARM 64</li></ul></td>
+      </tr>
+    </tbody>
+    </table>
+
+    > **Note:**
+    >
+    > - According to [CentOS Linux EOL](https://blog.centos.org/2023/04/end-dates-are-coming-for-centos-stream-8-and-centos-linux-7/), the upstream support for CentOS Linux 7 ends on June 30, 2024. TiDB ends the support for CentOS 7 starting from the 8.4 DMR version. It is recommended to use Rocky Linux 9.1 or a later version.
+    > - According to [CentOS Linux EOL](https://www.centos.org/centos-linux-eol/), the upstream support for CentOS Linux 8 ended on December 31, 2021. The upstream [support for CentOS Stream 8](https://blog.centos.org/2023/04/end-dates-are-coming-for-centos-stream-8-and-centos-linux-7/) ended on May 31, 2024. CentOS Stream 9 continues to be supported by the CentOS organization.
```
Comment (Oreoxmt) [inline hardware-and-software-requirements.md:66]: This line (L66) is missing from the Chinese documentation.

```markdown
    > - 根据 [CentOS Linux EOL](https://blog.centos.org/2023/04/end-dates-are-coming-for-centos-stream-8-and-centos-linux-7/)，CentOS Linux 7 的上游支持于 2024 年 6 月 30 日终止。从 8.4 DMR 版本开始，TiDB 已结束对 CentOS 7 的支持，建议使用 Rocky Linux 9.1 及以上的版本。
    > - 根据 [Red Hat Enterprise Linux Life Cycle](https://access.redhat.com/support/policy/updates/errata/#Life_Cycle_Dates)，Red Hat Enterprise Linux 7 的 Maintenance Support 于 2024 年 6 月 30 日终止。从 8.4 DMR 版本开始，TiDB 已结束对 Red Hat Enterprise Linux 7 的支持，建议使用 Rocky Linux 9.1 及以上的版本。
```

File: hardware-and-software-requirements.md [hunk 2]
```diff
@@ -17,85 +17,108 @@ As an open-source distributed SQL database with high performance, TiDB can be de
...
+      <tr>
+        <td>Oracle Enterprise Linux 8 or a later</td>
+        <td>x86_64</td>
+      </tr>
+      <tr>
+        <td>Ubuntu LTS 20.04 or later</td>
+        <td>x86_64</td>
+      </tr>
+      <tr>
+        <td>CentOS 8 Stream</td>
```
Comment (Oreoxmt) [inline hardware-and-software-requirements.md:92]:
```suggestion
        <td>CentOS Stream 8</td>
```

File: hardware-and-software-requirements.md [hunk 3]
```diff
@@ -17,85 +17,108 @@ As an open-source distributed SQL database with high performance, TiDB can be de
...
+        <td>x86_64</td>
+      </tr>
+    </tbody>
+    </table>
+
+    > **Note:**
+    >
+    > - For Oracle Enterprise Linux, TiDB supports the Red Hat Compatible Kernel (RHCK) and does not support the Unbreakable Enterprise Kernel provided by Oracle Enterprise Linux.
+    > - Support for Ubuntu 16.04 will be removed in future versions of TiDB. Upgrading to Ubuntu 18.04 or later is strongly recommended.
+     > - CentOS Stream 8 reach its [end of builds](https://blog.centos.org/2023/04/end-dates-are-coming-for-centos-stream-8-and-centos-linux-7/) on May 31, 2024.
```
Comment (qiancai) [inline hardware-and-software-requirements.md:117]:
```suggestion
     > - CentOS Stream 8 reaches its [end of builds](https://blog.centos.org/2023/04/end-dates-are-coming-for-centos-stream-8-and-centos-linux-7/) on May 31, 2024.
```

**Review by Oreoxmt — APPROVED** [review]

File: hardware-and-software-requirements.md [hunk 4]
```diff
@@ -17,85 +17,108 @@ As an open-source distributed SQL database with high performance, TiDB can be de
...
+        <td>x86_64</td>
+      </tr>
+    </tbody>
+    </table>
+
+    > **Note:**
+    >
+    > - For Oracle Enterprise Linux, TiDB supports the Red Hat Compatible Kernel (RHCK) and does not support the Unbreakable Enterprise Kernel provided by Oracle Enterprise Linux.
+    > - Support for Ubuntu 16.04 will be removed in future versions of TiDB. Upgrading to Ubuntu 18.04 or later is strongly recommended.
+     > - CentOS Stream 8 reaches its [end of builds](https://blog.centos.org/2023/04/end-dates-are-coming-for-centos-stream-8-and-centos-linux-7/) on May 31, 2024.
```
Comment (qiancai) [inline hardware-and-software-requirements.md:117]:
```suggestion
    > - CentOS Stream 8 reaches its [end of builds](https://blog.centos.org/2023/04/end-dates-are-coming-for-centos-stream-8-and-centos-linux-7/) on May 31, 2024.
```

**Pull request merged**